File: Data Structures & Algorithms/decode-ways/submission-1.cpp
class Solution {
public:
    int my(int i,string s,int n)
    {
        if(i>n)return 1;
        if(s[i]=='0')return 0;
        int a=my(i+1,s,n);
        int b=0;
        if(i+1<n)
        {
            if(s[i]=='1' || s[i]=='2' && s[i+1]<='6')
            {
                b=my(i+2,s,n);
            }
        }
        return a+b;
    }
    int numDecodings(string s) {
        int n=s.size();
        return my(0,s,n);
    }
};

File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        mp={}
        ans=[]
        for x in nums:
            mp[x]=mp.get(x,0)+1
        pq = []
        ans=[]
        for v,f in mp.items():
            heapq.heappush(pq, (-f, v))
        while k>0:
            f,v=heapq.heappop(pq)
            ans.append(v)
            k-=1
        return ans